services/agents/message_dispatcher.go

```diff
@@ -0,0 +1,38 @@
+package agents
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/api/agent"
+)
+
+type messageDispatcher struct {
```

Comment:
This type handles only responses, so maybe `responseDispatcher`?

And do we need it to be a separate type? Let's see if we can combine it with `Conn` after we do pmm-agent part.